src/Database/Driver/Mysql.php

```diff
@@ -71,7 +71,7 @@ class Mysql extends Driver
     /**
      * The schema dialect for this driver
      *
-     * @var \Cake\Database\Schema\MysqlSchemaDialect
+     * @var \Cake\Database\Schema\MysqlSchemaDialect|null
```

Comment:
I assume this is required because of the way it's initialized, but doesn't feel like it should be null.

Reply:
In the end it is important that for the consumer

    public function schemaDialect(): SchemaDialect

is true, which it is.
The change here seems "correct" from the way it is used internally, yeah.

Reply:
@othercorey The dialect instance instance isn't created from the constructor so if some method used the property directly without calling `schemaDialect()` it would be still null.